CRM/Contact/Selector.php

```diff
@@ -927,8 +927,22 @@ public function addActions(&$rows) {
     $mapMask = $mask & 4095;
 
     $links = self::links($this->_context, $this->_contextMenu, $this->_key);
+    $editlink = $links[CRM_Core_Action::UPDATE];  // Remember the Edit link
 
     foreach ($rows as $id => & $row) {
+      //
+      // CRM-12645 Check if we have permission to edit the record.  If we have, we enable the link, otherwise
+      // we don't.
+      if (CRM_Contact_BAO_Contact_Permission::allow($row['contact_id'], CRM_Core_Permission::EDIT)) {
```

Comment:
This function checks the logged in user for `edit all contacts` permission and returns TRUE irrespective of the `contact_id` passed. And if it doesn't have this permission edit link is already suppressed. So, I suspect this change won't be doing any extra handling for the links. Is there any use-case I'm missing here ?

AFAIK, links for the rows are combinedly based on logged in user permission and not for each contacts listed in the search result.